activerecord/lib/active_record/relation.rb

```diff
@@ -670,7 +670,17 @@ def values
     end
 
     def inspect
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      # A relation may be initialized with values that were not loaded from the
+      # database. Therefore, if there are values, they must be preserved by not
+      # loading from the database; else, use a new relation with a limit, to
+      # avoid loading an unbounded number of records into memory unnecessarily.
+      if values.any?
```

Comment:
How about
```patch
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      entries = 
+       if values.any?
+         records.take(num_to_take)
+        else
+          limit(num_to_take) # Not sure... it doesn't need to convert to an array
+        end
+      entries.map!(&:inspect)
```

Also I think will be cool to set that 11 and 10 as a constant (avoiding magic numbers)